flink-runtime/src/main/java/org/apache/flink/runtime/source/coordinator/ExecutorNotifier.java

```diff
@@ -153,7 +156,6 @@ public void close() throws InterruptedException {
             return;
         }
         // Shutdown the worker executor, so no more worker tasks can run.
-        workerExecutor.shutdownNow();
-        workerExecutor.awaitTermination(Long.MAX_VALUE, TimeUnit.SECONDS);
+        shutdownExecutorForcefully(workerExecutor, Duration.ofMillis(Long.MAX_VALUE));
```

Comment:
nit: I find it equally weird, similarly as in `SourceCoordinatorContext`, that we close something, given from the outside.